lib/mixin.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+function arrayUnion(a, b) {
+  let map = {};
+  let i = 0;
+  for (; i < a.length; i++) {
+    map[a[i]] = true;
+  }
+  for (i = 0; i < b.length; i++) {
+    map[b[i]] = true;
+  }
+  return Object.keys(map);
+}
+
+module.exports = function mixin(ext, parent) {
+  arrayUnion(Object.keys(ext), Object.keys(parent)).forEach(function(key) {
+    const extValue = ext[key];
+    const parentValue = parent[key];
+    
+    // If both the object and the parent define the same function
```

Comment:
this comment does not seem quite correct, as we do not seem to check for equality between extValue and parentValue merely that they are functions.